### PR TITLE
Parsing of newly introduced nodes, fixes for Firstula, and node storage by checksum_address.

### LINKED ISSUE #625: Whacky log parsing
Trying to log, as a warning, the message from an exception, I expect:

```
Not enough bytes to constitute
                    message types [(None, <class 'umbral.signing.Signature'>, 64, dict)] - need 64, got 0
```

Instead, I get:
```
Not enough bytes to constitute
                    message types [(None, <class 'umbral.signing.Signature'>, 64, dict)] - need 64, got 0
Temporarily disabling observer <twisted.logger._file.FileLogObserver object at 0x7fbe30ef0438> due to exception: [Failure instance: Traceback: <class 'KeyError'>: ''
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3267:run_code
<ipython-input-2-3454a5c79a45>:1:<module>
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py:238:warn
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py:144:emit
--- <exception caught here> ---
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_observer.py:131:__call__
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_file.py:50:__call__
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py:252:<lambda>
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py:203:eventAsJSON
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_flatten.py:108:flattenEvent
/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py:295:get_field
/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py:252:get_value
]
Temporarily disabling observer <twisted.logger._file.FileLogObserver object at 0x7fbe30ef0438> due to exception: [Failure instance: Traceback: <class 'KeyError'>: ''
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3267:run_code
<ipython-input-2-3454a5c79a45>:1:<module>
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py:238:warn
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py:144:emit
--- <exception caught here> ---
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_observer.py:131:__call__
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_file.py:50:__call__
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py:252:<lambda>
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py:203:eventAsJSON
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_flatten.py:108:flattenEvent
/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py:295:get_field
/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py:252:get_value
]
Temporarily disabling observer <twisted.logger._file.FileLogObserver object at 0x7fbe30ef0438> due to exception: [Failure instance: Traceback: <class 'KeyError'>: ''
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3267:run_code
<ipython-input-2-3454a5c79a45>:1:<module>
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py:238:warn
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py:144:emit
--- <exception caught here> ---
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_observer.py:131:__call__
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_file.py:50:__call__
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py:252:<lambda>
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py:203:eventAsJSON
/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_flatten.py:108:flattenEvent
/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py:295:get_field
/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py:252:get_value
]
Traceback (most recent call last):
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3267, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-3454a5c79a45>", line 1, in <module>
    self.log.warn(e.args[0])
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py", line 238, in warn
    self.emit(LogLevel.warn, format, **kwargs)
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_logger.py", line 144, in emit
    self.observer(event)
--- <exception caught here> ---
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_observer.py", line 131, in __call__
    observer(event)
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_file.py", line 50, in __call__
    text = self.formatEvent(event)
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py", line 252, in <lambda>
    lambda event: u"{0}{1}\n".format(recordSeparator, eventAsJSON(event))
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_json.py", line 203, in eventAsJSON
    flattenEvent(event)
  File "/home/jmyles/.local/share/virtualenvs/nucypher-EzeC4Bqx/lib/python3.7/site-packages/twisted/logger/_flatten.py", line 108, in flattenEvent
    field = aFormatter.get_field(fieldName, (), event)
  File "/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py", line 295, in get_field
    obj = self.get_value(first, args, kwargs)
  File "/home/jmyles/projects/pyenv/versions/3.7.0/lib/python3.7/string.py", line 252, in get_value
    return kwargs[key]
builtins.KeyError: ''
```

### LINKED ISSUE #630: Unhandled 500 during network bootstraping
`FleetStateTracker.snapshot` fails when `self.checksum` is `NO_KNOWN_NODES`  

~ Firstula!

### LINKED ISSUE #654: Set wallet address on certificate (and subsequently glean it from the same)
Lively discussion starting here: https://discordapp.com/channels/411401661714792449/411401661714792451/531897652288421889

### PR DESCRIPTION
**Pull request by jMyles**
Many small fixes based on mistakes and glitches observed in the first iteration of NuFT.  

Also:

* Fixes #625 - Logs are now parsed so as to avoid an exception if log message contains curlies. 
* Fixes #630 - Handling scenario where node has to serialize `NO_KNOWN_NODES`
* Fixes #654 - Certs are now saved by wallet address; wallet address is included in cert.

### CODE REVIEW

File: nucypher/crypto/api.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def ecdsa_verify(message: bytes,
 
 
 def generate_self_signed_certificate(host: str,
+                                     checksum_address: str,
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/crypto/api.py]: This is fixed upstream in PR #660

File: nucypher/network/nodes.py [hunk 2]
```diff
@@ -688,10 +699,19 @@ def learn_from_teacher_node(self, eager=True):
             if GLOBAL_DOMAIN not in self.learning_domains:
                 if not self.learning_domains.intersection(node.serving_domains):
                     continue  # This node is not serving any of our domains.
+
+            # First, determine if this is an outdated representation of an already known node.
+            with suppress(KeyError):
```
Comment (tuxxy) [inline nucypher/network/nodes.py]: :/

File: nucypher/crypto/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jMyles) [reply nucypher/crypto/api.py]: Downstream, yeah?

[the remark this answers is not in the document]
Comment (jMyles) [reply nucypher/crypto/api.py]: (ie, we're going to merge this first, right?)

[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/crypto/api.py]: Whoops! Yes.

**Pull request merged**